src/marketplace/Apps.Service/BusinessLogic/AppsBusinessLogic.cs

```
@@ -375,4 +376,65 @@ private static async Task<IEnumerable<OfferDescriptionData>> ValidateAndGetAppDe
         }
         return result.OfferDescriptionDatas;
     }
+
+    /// <inheritdoc />
+    public async Task CreatOfferAssignedAppLeadImageDocumentByIdAsync(Guid appId, string iamUserId, IFormFile document, CancellationToken cancellationToken)
+    {
+        var appLeadImageContentTypes = new []{ "image/jpeg","image/png" };
+        if (!appLeadImageContentTypes.Contains(document.ContentType))
+        {
+            throw new UnsupportedMediaTypeException($"Document type not supported. File with contentType :{string.Join(",", appLeadImageContentTypes)} are allowed.");
+        }
+
+        var offerRepository = _portalRepositories.GetInstance<IOfferRepository>();
+        var result = await offerRepository.GetOfferAssignedAppLeadImageDocumentsByIdAsync(appId, iamUserId, OfferTypeId.APP).ConfigureAwait(false);
```

result could be default here if no offer for the given offer type is found. Please add a check for that and throw a not found exception